src/worker/mod.rs

```diff
@@ -75,43 +73,56 @@ pub struct Worker {
     /// Has shutdown been called?
     shutdown_requested: watch::Receiver<bool>,
     shutdown_sender: watch::Sender<bool>,
+
+    metrics: MetricsContext,
 }
 
 impl Worker {
     pub(crate) fn new(
         config: WorkerConfig,
         sticky_queue_name: Option<String>,
         sg: Arc<GatewayRef>,
+        metrics: MetricsContext,
     ) -> Self {
+        metrics.worker_registered();
```

Comment:
Should this include the task queue?

Reply:
Already does from higher context. A lot of that is nonobvious though. There is a way I can fix it with clever typing but I didn't want to spend quite that much time initially. There is a note about how it can be done in the future if needed